src/uct/base/uct_mem.c

```diff
@@ -138,10 +135,22 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
                 ucs_assert(memh != UCT_MEM_HANDLE_NULL);
                 mem->md       = md;
                 mem->mem_type = md_attr.cap.access_mem_type;
+                // TODO: Need to fix this because mem->type is defined as type
+                // of memory allocated but currently it's value is set to
+                // access_mem_type instead of param->mem_type
                 mem->memh     = memh;
                 goto allocated;
 
             }
+
+            if (param->mem_type != UCS_MEMORY_TYPE_HOST) {
```

Comment:
what if the first method is mmap but mem type is not HOST?

Reply:
If mem_type requested is not HOST then I guess we should first go through the methods and check if METHOD_MD exists. If not, we should error out. If it does exist, then we go through all MDs and error out if none of the MDs support the requested mem_type.

Reply:
if every allocation method which is not MD, we should skip it if the memory type is not HOST